File: nsi-platform/frontend/android/app/src/main/java/com/nsi/app/models/AppState.kt
package com.nsi.app.models

import androidx.lifecycle.ViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.serialization.Serializable

data class UserInfo(val nickName: String = "default")

@Serializable
data class ProfileInput(
    val age: Int,
    val gender: String,
    val householdRegionCode: String,
    val currentResidenceCode: String,
    val employmentStatus: String,
    val socialSecurityYears: Int,
    val hasChildren: Boolean,
)

@Serializable
data class PlanInput(
    val age: Int,
    val gender: String,
    val employment: String,
    val contributionYears: Int,
    val currentBalance: Double,
    val monthlyBudget: Double,
    val localAvgSalary: Double,
)

data class Scheme(
    val name: String,
    val baseSalary: Int,
    val monthlyCost: Double,
    val projectedPension: Double,
    val annualSubsidy: Double,
)

data class PlanResult(
    val planId: String,
    val recommendedSchemes: List<Scheme>,
)

data class AppUiState(
    val userInfo: UserInfo = UserInfo("default"),
    val currentCity: String = "上海",
    val currentCityCode: String = "310000",
    val profileData: ProfileInput? = null,
    val planInput: PlanInput? = null,
    val planResult: PlanResult? = null,
)

class AppViewModel : ViewModel() {
    private val _uiState = MutableStateFlow(AppUiState())
    val uiState: StateFlow<AppUiState> = _uiState.asStateFlow()

    fun setUserInfo(info: UserInfo) {
        _uiState.value = _uiState.value.copy(userInfo = info)
    }

    fun setCity(name: String, code: String) {
        _uiState.value = _uiState.value.copy(currentCity = name, currentCityCode = code)
    }

    fun setProfileData(data: ProfileInput) {
        _uiState.value = _uiState.value.copy(profileData = data)
    }

    fun setPlanInput(input: PlanInput) {
        _uiState.value = _uiState.value.copy(planInput = input)
    }

    fun setPlanResult(result: PlanResult) {
        _uiState.value = _uiState.value.copy(planResult = result)
    }
}

File: nsi-platform/frontend/ios/Models/AppState.swift
import Foundation
import CoreLocation

@MainActor
class AppState: ObservableObject {
    @Published var userInfo: UserInfo? = UserInfo(nickName: "default")
    @Published var currentCity: String = "上海"
    @Published var currentCityCode: String = "310000"
    @Published var planResult: PlanResult?
    @Published var profileData: ProfileInput?
    @Published var planInput: PlanInput?

    struct UserInfo {
        let nickName: String
    }

    struct ProfileInput: Codable {
        let age: Int
        let gender: String
        let householdRegionCode: String
        let currentResidenceCode: String
        let employmentStatus: String
        let socialSecurityYears: Int
        let hasChildren: Bool
    }

    struct PlanInput: Codable {
        let age: Int
        let gender: String
        let employment: String
        let contributionYears: Int
        let currentBalance: Double
        let monthlyBudget: Double
        let localAvgSalary: Double
    }

    struct PlanResult: Codable {
        let planId: String
        let recommendedSchemes: [Scheme]
    }

    struct Scheme: Codable, Identifiable {
        var id: String { name }
        let name: String
        let baseSalary: Int
        let monthlyCost: Double
        let projectedPension: Double
        let annualSubsidy: Double
    }
}
